### ISSUE TITLE
feat(eval-report): flatten_report_to_metrics — support multi-series and multi-point charts

### ISSUE THREAD

**Issue by sallycr** [michelangelo-ai/michelangelo#1258]
## Problem

`flatten_report_to_metrics()` in `michelangelo/workflow/tasks/functions/eval_report_sinks/base.py` currently only extracts **scalar** values from an `EvaluationReport`: charts with exactly one series and one data point.

Two categories of charts are silently skipped today:

1. **Multi-series charts** (e.g. train loss vs. val loss on the same chart, or A/B comparison overlays) — charts where `len(series) != 1`.
2. **Multi-point series** (time-series / learning curves logged per epoch/step) — charts where `len(data_points) != 1`.

Both cases log a `DEBUG` message but produce no output in the returned dict. Callers using `mlflow.log_metrics()`, `wandb.log()`, or Comet have no way to know data was dropped.

## Impact

- **Time-series** charts (loss curves, accuracy over epochs) are the most common output of training jobs — these are universally skipped today.
- MLflow `log_metric(key, value, step=step)`, W&B `wandb.log({key: value}, step=step)`, Comet `log_metric(key, value, step=step)` all accept a `step` argument for time-series. The `x` field in each `DataPoint` maps directly to this, but is currently dropped.
- Multi-series charts cannot be disambiguated by title alone; they need either a per-series key suffix or caller-controlled name mapping.

### Multi-point series
Emit one entry per data point with a `step` index:

```python
# current behaviour (single scalar)
{"accuracy": 0.95}

# proposed: multi-point emits step-keyed entries or calls log_metric with step
{"accuracy/step_0": 0.80, "accuracy/step_1": 0.85, "accuracy/step_2": 0.95}
# — or —
mlflow.log_metric("accuracy", value, step=x)  # using DataPoint.x field
```

Two sub-options:
- **Flat dict with step suffix** — compatible with the current return type (`dict[str, float]`), no API change.
- **Structured return type** — return `dict[str, list[tuple[float, float]]]` (value, step), breaking change but richer.

### Multi-series charts
Options:
- Use `series[i].label` (if populated) as a key suffix: `f"{chart_title}/{series_label}"`.
- Skip silently with a caller-visible warning (current behaviour, but at `WARNING` not `DEBUG`).

### Step field
`DataPoint.x` is the step/epoch axis value. Currently ignored. Should be surfaced so callers can pass it to `mlflow.log_metric(step=)` etc.

## Acceptance criteria

- [ ] `flatten_report_to_metrics` (or a new helper) supports multi-point series without data loss
- [ ] The `x` (step) field is accessible to the caller
- [ ] Multi-series charts are handled (either via label suffix or documented skip)
- [ ] Existing scalar behaviour is unchanged (no regression)
- [ ] Helper docstring updated with examples for all three chart shapes

## Related

- `flatten_report_to_metrics` is used by custom sinks targeting MLflow, W&B, Comet, Neptune
- `EvalReportSink.write()` contract: helper is provided for convenience, not called by the built-in sinks
- TODO markers in code: `base.py` lines with `# TODO(v2):`
